lexer.go

```diff
@@ -274,6 +274,26 @@ func (l *Lexer) Lex(lval *yySymType) (token int) {
 		return int('|')
 	}
 
+	switch ch := l.ch; ch {
+	case '"', '`', '[':
+		closingChar := map[byte]byte{
+			'"': '"',
+			'`': '`',
+			'[': ']',
+		}
+		l.readByte() // consume opening char
+		literal := l.readIdentifier()
+		if l.ch != closingChar[ch] {
+			l.literal = literal
+			return ERROR
+		}
+		l.readByte() // consume closing char
+
+		l.literal = literal
+		lval.bytes = literal
```

Comment:
This is how you pass (`lval.bytes = literal`) the identifier bytes to the parser. That's how those dollar sign variables from the grammar ($) have values.